### PR TITLE
Tests for Capture, Choice, TalkStep

### LINKED ISSUE #82: Edited configurations removing all backend errors!
1. went to backend/pom.xml, right clicked on the code > add maven project
2. add 				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<version>${project.parent.version}</version> 
in pom.xml (backend)

### PR DESCRIPTION
**Pull request by Lemiex**
Test line coverage:
- Choice Step 100%
- Capture 100%
- TalkStep 71% line coverage. Since TalkStep is an abstract class there were some methods and attributes I couldn't test. Let me know if anyone knows how to go about this

Changed magic numbers in Choice and Capture Step to be variables from ScoreStandards

### CODE REVIEW

**Comment by Lemiex** [comment]
The java test fails because we haven't merged Zoey's PR with ScoreStandards have not been merged

File: backend/src/test/java/com/speakfluid/backend/entities/CaptureStepTests.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                            entry("which day", ScoreStandards.highMatch),
+                            entry("when", ScoreStandards.mediumMatch),
+                            entry("where", ScoreStandards.mediumMatch),
+                            entry("time", ScoreStandards.mediumMatch),
+                            entry("date", ScoreStandards.lowMatch),
+                            entry("day", ScoreStandards.lowMatch)),
+                    Map.ofEntries(entry("address", ScoreStandards.highMatch),
+                            entry("zip code", ScoreStandards.highMatch),
+                            entry("location", ScoreStandards.highMatch)),
+                    Map.ofEntries( entry("name", ScoreStandards.mediumMatch),
```
Comment (ZZBP) [inline backend/src/test/java/com/speakfluid/backend/entities/CaptureStepTests.java]: Why are the keywords included in the tests? Are they used anywhere? I might just be blind T_T

File: backend/src/main/java/com/speakfluid/backend/entities/ChoiceStep.java [hunk 2]
```diff
@@ -54,27 +73,11 @@ public void runAnalysis(Dialogue dialogue) {
         }
         // User messages
         for (Object userMessage : dialogue.getUserMessage()){
-            countMatchKeywords((Message) userMessage, choiceKeyWordsUser);
+            scoreAccumulator += countMatchKeywords((Message) userMessage, choiceKeyWordsUser);
             if(calculateMsgLength((Message) userMessage) <= 3) {
-                scoreAccumulator += 7;
+                scoreAccumulator += 3.0;
```
Comment (ZZBP) [inline backend/src/main/java/com/speakfluid/backend/entities/ChoiceStep.java:78]: Here you can use ScoreStandards.mediumMatch to replace 3.0

File: backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java [hunk 3]
```diff
@@ -92,9 +141,10 @@ public void isEmail(Message message){
      */
     public void isZipCode(Message message){
         for(String word : message.getMessage().split(" ")){
-            if(word.matches( "^\\d{5}([-+]?\\d{4})?$")
-                    || word.matches( "/^[A-Za-z]\\d[A-Za-z][ -]?\\d[A-Za-z]\\d$/")){
-                scoreAccumulator += 10;
+            if(word.matches( "^\\d{5}([-+]?\\d{4})?$")  //american postal code
+                    || word.matches( "^(?!.*[DFIOQU])[A-VXY][0-9][A-Z]●?[0-9][A-Z][0-9]$") //full canadian postal code
+                    || word.matches( "^(?!.*[DFIOQU])[A-VXY][0-9][A-Z]")){ //half a canadian (such as M5S) since people often put a space in between the two parts
+                scoreAccumulator += 3.0;
```
Comment (ZZBP) [inline backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java]: The 3.0 here can also be replaced by ScoreStandards.mediumMatch

File: backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java [hunk 4]
```diff
@@ -64,7 +113,7 @@ public CaptureStep(){
     public void hasNumbers(Message message){
         for(String word : message.getMessage().split(" ")){
             if(word.matches(".*[0-9].*")){
-                scoreAccumulator += 3;
+                scoreAccumulator += 1.0;
```
Comment (ZZBP) [inline backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java]: 1.0 can be replaced by ScoreStandards.lowMatch

**Review by ZZBP — CHANGES_REQUESTED** [review]
Hey Sarah! Appreciate all your amazing work writing throughout tests and incorporating the step entities with the ScoreStandards! Just a few quick notes: 
1. For encodings.xml and misc.xml, are those wanted deletions?
2. Within the step entities, all the numerical scores used in methods such as 1.0, and 3.0 can now be replaced by ScoreStandards.low/medium/highMatch. 
3. Why are the specific keywords included in the tests? Are they used? If so, lmk, since I should include them as well for the tests of the other step entities!

**Review by auroraborie — APPROVED** [review]
Hi Sarah great work on the tests!
I also saw you returned the lowercase string now in TalkStep which is a good detail to implement as I remember our standards were to assume the messages were all lower case. I also think the tests scored fairly high in the score accumulator attribute when given a targeted dialogue which is good! Zoey brought up the point of using the ScoreStandard class now more prevalently in our code which is a good idea. I think the rest looks good.

File: backend/src/test/java/com/speakfluid/backend/entities/CaptureStepTests.java (hunk 1 above)
Comment (Lemiex) [reply backend/src/test/java/com/speakfluid/backend/entities/CaptureStepTests.java]: Good catch, i didn't need them!

File: backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java (hunk 4 above)
Comment (Lemiex) [reply backend/src/main/java/com/speakfluid/backend/entities/CaptureStep.java]: Yessir no magic numbers

**Review by ZZBP — APPROVED** [review]
Thanks for the changes!!!

**Pull request merged**